sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -6,35 +6,51 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/centralsensor"
 	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
 type nodeScanHandlerImpl struct {
 	nodeScans <-chan *storage.NodeScanV2
-	toCentral chan *central.MsgFromSensor
+	toCentral <-chan *central.MsgFromSensor
+	// toCentralMux prevents the race condition between Start() [writer] and ResponsesC() [reader]
+	toCentralMux *sync.Mutex
```

Comment:
also, it looks like this mutex it used for more than just `toCentral`, but rather to ensure `Start()` and `ResponseC` do are not run in parallel/concurrently. Maybe this can just be called `lock`? (Also I think mux typically implies multiplexer instead of mutex)